File: lgf-tapper-api/src/main/java/com/lgf/tapper/domain/Consumption.java
package com.lgf.tapper.domain;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.format.annotation.DateTimeFormat;

import java.sql.Date;
import java.sql.Timestamp;
import java.text.DateFormat;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

@Data
@Setter
public class Consumption {

    @Id
    private String id;
    private LocalDateTime createdOn;
    private Product product;
    private ClubMember clubMember;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    public LocalDateTime getCreatedOn() {
        return createdOn;
    }


}

File: lgf-tapper-ui/src/app/domain/consumption.ts
import { Product } from './product';
import { ClubMember } from './clubMember';

export class Consumption {
constructor(
  public id: number,
  public date: Date,
  public product: Product,
  public clubMember: ClubMember,
  public photoBase64Encoded: string) { }
}
